source/slang/slang-doc.cpp

```diff
@@ -0,0 +1,972 @@
+// slang-doc.cpp
+#include "slang-doc.h"
+
+#include "../core/slang-string-util.h"
+
+namespace Slang {
+
+/* TODO(JS):
+
+* If Decls hand SourceRange, then we could use the range to simplify getting the Post markup, as will be trivial to get to the 'end'
+* Need to handle preceeding * in some markup styles
+* If we want to be able to disable markup we need a mechanism to do this. Probably define source ranges.
+
+* Need a way to take the extracted markup and produce suitable markdown
+** This will need to display the decoration appropriately
+*/
+
+class MarkupExtractor 
```

Comment:
For terminology, should this be a `DocumentationCommentExtractor` (or `DocCommentExtractor` if we need to abbreviate)? It seems like there is potential for confusion (like I immediately had when reading the type name) between "markup" as in "we have marked up these declarations with documentation comments" vs. "markup" as in "the markup language we are using for documentation comments is markdown."